UI/Pages/Shortlist.cshtml

```diff
@@ -21,13 +21,64 @@
             @if (count > 0)
             {
                 var dataQueryString = Model.Data.ToQueryString();
+
+                <input asp-for="Data.From" type="hidden" />
+                <input asp-for="Data.Postcode" type="hidden" />
+                @if (Model.Data.KeyStages != null)
+                {
+                    @for(int i = 0; i < Model.Data.KeyStages.Count(); i++)
+                    {
+                        <input asp-for="Data.KeyStages[i]" type="hidden" />
```

Comment:
Dereference of a possibly null reference. I suggest the below change

` <input asp-for="Data.KeyStages![i]" type="hidden" />`

Reply:
Done - as mentioned, this wasn't flagging as an error in VS for some reason!